### ISSUE TITLE
Windows 11, Flutter app (Target: Android) doesn't Start, Build Fails on Windows 11, however works fine on MacOS

### ISSUE THREAD

**Issue by DuaRobin** [flutter/flutter#99223]
## Steps to Reproduce

1. I'm on windows 11, Using Flutter 2.10.2, Created a new application with built in template using 'flutter create hello_flutter'.
2. Started Android Emulator and Executed flutter run.

**Expected results:**

- App should have been started in Android.

**Actual results:**

- App never started because of Build error.

<details>
<summary>Code sample</summary>

```
Template Project w/ Flutter 2.10.2 and NO Changes
```
</details>

<details>
  <summary>Logs</summary>

```
Launching lib\main.dart on sdk gphone64 x86 64 in debug mode...

FAILURE: Build failed with an exception.

* What went wrong:
Could not open settings generic class cache for settings file 'C:\Users\<user>\source\repos\hello_flutter\android\settings.gradle' (C:\Users\<user>\.gradle\caches\6.7\scripts\1fnwrr8g4rohfp291nvlxj5qe).
> BUG! exception in phase 'semantic analysis' in source unit '_BuildScript_' Unsupported class file major version 61

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.
* Get more help at https://help.gradle.org

BUILD FAILED in 504ms
Exception: Gradle task assembleDebug failed with exit code 1
Exited (sigterm)

```
```
PS C:\Users\<user>\source\repos\hello_flutter> flutter doctor -v
[✓] Flutter (Channel stable, 2.10.2, on Microsoft Windows [Version 10.0.22000.527], locale en-US)
    • Flutter version 2.10.2 at C:\Users\<user>\AppData\Local\Flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 097d3313d8 (8 days ago), 2022-02-18 19:33:08 -0600
    • Engine revision a83ed0e5e3
    • Dart version 2.16.1
    • DevTools version 2.9.2

[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
    • Android SDK at C:\Program Files (x86)\Android\android-sdk
    • Platform android-31, build-tools 30.0.3
    • Java binary at: C:\Program Files\Eclipse Adoptium\jdk-17.0.1.12-hotspot\bin\java
    • Java version OpenJDK Runtime Environment Temurin-17.0.1+12 (build 17.0.1+12)
    • All Android licenses accepted.

[✗] Chrome - develop for the web (Cannot find Chrome executable at .\Google\Chrome\Application\chrome.exe)
    ! Cannot find Chrome. Try setting CHROME_EXECUTABLE to a Chrome executable.

[✓] Visual Studio - develop for Windows (Visual Studio Community 2022 17.1.0)
    • Visual Studio at C:\Program Files\Microsoft Visual Studio\2022\Community
    • Visual Studio Community 2022 version 17.1.32210.238
    • Windows 10 SDK version 10.0.19041.0

[!] Android Studio (not installed)
    • Android Studio not found; download from https://developer.android.com/studio/index.html
      (or visit https://flutter.dev/docs/get-started/install/windows#android-setup for detailed instructions).

[✓] VS Code (version 1.64.2)
    • VS Code at C:\Users\<user>\AppData\Local\Programs\Microsoft VS Code
    • Flutter extension version 3.34.0

[✓] Connected device (3 available)
    • sdk gphone64 x86 64 (mobile) • emulator-5554 • android-x64    • Android 12 (API 31) (emulator)
    • Windows (desktop)            • windows       • windows-x64    • Microsoft Windows [Version 10.0.22000.527]
    • Edge (web)                   • edge          • web-javascript • Microsoft Edge 98.0.1108.62

[✓] HTTP Host Availability
    • All required HTTP hosts are available

! Doctor found issues in 2 categories.
```
</details>

<details>
<summary>Additional Info</summary>

```
Note:
I tried Exactly SAME Emulator, with Same Android SDK on MacOS, and Flutter Build was just fine there, and app worked. 
On Windows however it's failing with above error.
```

</details>

**Comment by CodeDoctorDE** [comment]
Try using java 11 (you can check it via `java -version`)
see https://stackoverflow.com/a/67841975

**Comment by darshankawar** [comment]
@DuaRobin 
The error `BUG! exception in phase 'semantic analysis' in source unit '_BuildScript_' Unsupported class file major version 61` seems to be coming from gradle which doesn't seem to be compatible with jdk version you are using. Try to downgrade your jdk version to a version less than what you are using and see if it helps.

See this similar issue for your reference: https://stackoverflow.com/questions/66893535/why-my-flutter-app-dont-run-bug-exception-in-phase-semantic-analysis-in-sou

Closing from here, as this isn't a Flutter issue. If you disagree, write in comments and I'll reopen it.

**Comment by DuaRobin** [comment]
@darshankawar I'm using JDK 11 on both MAC & WIN11, Mac works and Win 11 doesn't. I don't think it's related to JDK compatibility here. Also Win 11 Android Emulator for other apps works just fine too. I think this should remain open for some additional research.

**Comment by CodeDoctorDE** [comment]
`flutter doctor` says you are using java 17

**Comment by DuaRobin** [comment]
> `flutter doctor` says you are using java 17

ok, I notice my machine had both JDK 17 and 11, however on terminal java --version was returning JDK 11. 
I now have changed Java_HOME to point to JDK 11, and also flutter doctor points to same JDK. But Build is still failing with a different error now, any inputs?

<details>
<summary>flutter doctor</summary>

```
PS C:\Users\<user>> flutter doctor -v
[✓] Flutter (Channel stable, 2.10.2, on Microsoft Windows [Version 10.0.22000.527], locale en-US)
    • Flutter version 2.10.2 at C:\Users\<user>\AppData\Local\Flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 097d3313d8 (10 days ago), 2022-02-18 19:33:08 -0600
    • Engine revision a83ed0e5e3
    • Dart version 2.16.1
    • DevTools version 2.9.2

[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
    • Android SDK at C:\Program Files (x86)\Android\android-sdk
    • Platform android-31, build-tools 30.0.3
    • Java binary at: C:\Program Files\Microsoft\jdk-11.0.12.7-hotspot\bin\java
    • Java version OpenJDK Runtime Environment Microsoft-25199 (build 11.0.12+7)
    • All Android licenses accepted.

[✗] Chrome - develop for the web (Cannot find Chrome executable at .\Google\Chrome\Application\chrome.exe)
    ! Cannot find Chrome. Try setting CHROME_EXECUTABLE to a Chrome executable.

[✓] Visual Studio - develop for Windows (Visual Studio Community 2022 17.1.0)
    • Visual Studio at C:\Program Files\Microsoft Visual Studio\2022\Community
    • Visual Studio Community 2022 version 17.1.32210.238
    • Windows 10 SDK version 10.0.19041.0

[!] Android Studio (not installed)
    • Android Studio not found; download from https://developer.android.com/studio/index.html
      (or visit https://flutter.dev/docs/get-started/install/windows#android-setup for detailed instructions).

[✓] VS Code (version 1.64.2)
    • VS Code at C:\Users\<user>\AppData\Local\Programs\Microsoft VS Code
    • Flutter extension version 3.34.0

[✓] Connected device (3 available)
    • sdk gphone64 x86 64 (mobile) • emulator-5554 • android-x64    • Android 12 (API 31) (emulator)
    • Windows (desktop)            • windows       • windows-x64    • Microsoft Windows [Version 10.0.22000.527]
    • Edge (web)                   • edge          • web-javascript • Microsoft Edge 98.0.1108.62

[✓] HTTP Host Availability
    • All required HTTP hosts are available

! Doctor found issues in 2 categories.
```
</details>

<details>
<summary>Current Error</summary>

```
Launching lib\main.dart on sdk gphone64 x86 64 in debug mode...
Checking the license for package Android SDK Build-Tools 29.0.2 in C:\Program Files (x86)\Android\android-sdk\licenses
License for package Android SDK Build-Tools 29.0.2 accepted.
Preparing "Install Android SDK Build-Tools 29.0.2 (revision: 29.0.2)".
Warning: Failed to read or create install properties file.

FAILURE: Build failed with an exception.
* What went wrong:
Could not determine the dependencies of task ':app:compileDebugJavaWithJavac'.
> Failed to install the following SDK components:
      build-tools;29.0.2 Android SDK Build-Tools 29.0.2
  Install the missing components using the SDK manager in Android Studio.
* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 998ms
Exception: Gradle task assembleDebug failed with exit code 1
Exited (sigterm)
```

**Comment by darshankawar** [comment]
@DuaRobin 
As the error indicates `Install the missing components using the SDK manager in Android Studio.`, you'll need to install the missing build tools using SDK manager in AS.

**Comment by DuaRobin** [comment]
@darshankawar @CodeDoctorDE 

okay, just for the documentation, here is the root cause and my solution:

- VS installs Android SDK to a path: `C:\Program Files (x86)\Android` by default and does NOT even give an option to change it. 
- Now Flutter says, not to put anything on Program Files because it needs elevated access, and it actually fails to add SDK version 29 to same Directory where VS has installed other SDK versions (let say 31). 
- However, Android Studio on the other hand installs SDK to a path:  `C\:Users\<name>\AppData\Local\Android\Sdk` that Flutter works just fine with. **_What this means is if a user/developer wants to develop Android Apps and likes both VS and Flutter, user has to install Android Studio too which is unnecessary if VS has already installed Android SDK's. Yes I know Flutter installation Guide includes Android studio but we can manage without it._**

> **_What was done to Fix it_**? : 

- [x] I wanted to keep my machine tidy and did NOT want to install Android Studio unless it was absolutely needed. But I had to at least _Once_ to even find the root cause.
- [x] Flutter Config was then manually changed to point to correct SDK Path [the one that comes with Android studio] using command: `flutter config --android-sdk <path-to-your-android-sdk-path>`. 
- [x] Flutter since ran fine also installed SDK version 29 automatically to same SDK directory. 
- [x] I manually pasted SDK version 29 to VS SDK path and Un-Installed everything related to Android Studio including SDK Paths. I wish VS SDK Manager tool could have shown older SDK versions for us to choose and install too. Then I wouldn't have needed to paste it manually.
- [x] Flutter config was changed once more to switch back to VS SDK path, it now works fine using VS SDK Path too.

> A Suggestion was submitted to Microsoft/VS Team as well to pick same directory that Android Studio chooses, Hoping they'd consider it.
